linalg/ode.hpp

```diff
@@ -530,27 +383,173 @@ class ESDIRK33Solver : public ODESolver
 class GeneralizedAlphaSolver : public ODESolver
 {
 protected:
-   mutable Vector xdot,k,y;
+   mutable Vector k,y;
    double alpha_f, alpha_m, gamma;
-   int  nstate;
 
    void SetRhoInf(double rho_inf);
    void PrintProperties(std::ostream &out = mfem::out);
 public:
 
    GeneralizedAlphaSolver(double rho = 1.0) { SetRhoInf(rho); };
+   void Init(TimeDependentOperator &f_) override;
+   void Step(Vector &x, double &t, double &dt) override;
+
+};
 
+/** An explicit Adams-Bashforth method. */
+class AdamsBashforthSolver : public ODESolver
+{
+private:
+   const double *a;
+   int stages;
+   double dt_;
+
+protected:
+   ODESolver* RKsolver;
```

Comment:
Here and in `AdamsMoultonSolver`, `RKsolver` can be a `unique_ptr` (and so the destructor can be removed)

Reply:
Thanks for the pointer :). 

Done!